astropy/cosmology/tests/test_core.py

```diff
@@ -50,7 +50,8 @@
 
 invalid_zs = [
     (None, TypeError),  # wrong type
-    (4 * u.MeV, u.UnitConversionError),  # wrong unit
+    (4 * u.MeV, (u.UnitConversionError, TypeError)),  # scalar with wrong unit
```

Comment:
Would a generic `u.UnitsError` be more useful? (There are a few combinations of `UnitsError` with `ValueError` and `TypeError`)

Reply:
The TypeError is actually for the c methods. The UnitConversionError is the right one for any python method.